misp_stix_converter/converters/buildMISPAttribute.py

```diff
@@ -23,55 +23,170 @@
+            mispEvent.add_attribute('sha1', six.text_type(obj.sha1), comment=pkg.title or None)
 
-        if i.properties.sha256:
-            if len(i.properties.sha256) == 64:
-                mispEvent.add_attribute('sha256', six.text_type(i.properties.sha256), comment=pkg.title or None)
+    if obj.sha256:
+        if len(obj.sha256) == 64:
+            mispEvent.add_attribute('sha256', six.text_type(obj.sha256), comment=pkg.title or None)	
+    
+    # Added support for SHA512 (DB)
+    if obj.sha512:
+        if len(obj.sha512) == 128:
+            mispEvent.add_attribute('sha512', six.text_type(obj.sha512), comment=pkg.title or None)
+            
+    if objRel:
+        if pkg.object_.related_objects:
+            parseRelated(pkg.object_.related_objects, mispEvent, pkg)
+    
+    return mispEvent
+    
+# Dedicated to Address Object (DB)
+def buildAddressAttribute(obj, mispEvent, pkg, objRel=False):
 
+    if obj.is_source:
+        mispEvent.add_attribute('ip-src', six.text_type(obj.address_value), comment=pkg.title or None)
+    
+    elif obj.is_destination:
+        mispEvent.add_attribute('ip-dst', six.text_type(obj.address_value), comment=pkg.title or None)
+    else:
+    # We don't know, first check if it's an IP range
+        if hasattr(obj, "condition") and obj.condition:
+            if obj.condition == "InclusiveBetween":
+            # Ok, so it's a range. hm. Shall we add them seperately#comma#or together?
+                mispEvent.add_attribute('ip-dst', six.text_type(obj.address_value[0]))
+                mispEvent.add_attribute('ip-dst', six.text_type(obj.add_attribute[1]))
+            elif obj.condition == "Equals":
+                mispEvent.add_attribute('ip-dst', six.text_type(obj.address_value), comment=pkg.title or None)
+            else:
+            # Unknown condition
+                pass
+        else:
+            # Don't have anything to go on
+            mispEvent.add_attribute('ip-dst', six.text_type(obj.address_value), comment=pkg.title or None)
+            
+    if objRel:
+        if pkg.object_.related_objects:
+            parseRelated(pkg.object_.related_objects, mispEvent, pkg)
+    
     return mispEvent
-#Added by Davide Baglieri (aka davidonzo)
+
+# Dedicated to EmailMessage (DB)
+def buildEmailMessageAttribute(obj, mispEvent, pkg, objRel=False):
+    if obj.header:
+        # We have a header, can check for to/from etc etc
+        if obj.header.from_:
+            mispEvent.add_attribute('email-src', six.text_type(obj.header.from_.address_value), comment=pkg.title or None)
+        if obj.header.to:
+            for mail in obj.header.to:
+                mispEvent.add_attribute('email-dst', six.text_type(mail.address_value), comment=pkg.title or None)
+        if obj.header.subject:
+            mispEvent.add_attribute('email-subject', six.text_type(obj.header.subject), comment=pkg.title or None)
+            
+    if obj.attachments:                    
```

Comment:
you can squish this down to `if obj.attachments and pkg.object_.related_objects`

ew, `object_` is an icky way to get around python's reserved words. It's STIX's fault :<